lib/ood_core/job/queue_info.rb

```diff
@@ -26,6 +28,8 @@ class OodCore::Job::QueueInfo
   def initialize(**opts)
     @name = opts.fetch(:name, 'unknown')
     @qos = opts.fetch(:qos, [])
+    @allow_qos = opts.fetch(:allow_qos, nil)
```

Comment:
Instead of `nil` we should use an empty array `[]` so that we don't get nil errors.